packages/react-components/react-tree/stories/Sticky/StickyTree.stories.tsx

```diff
@@ -0,0 +1,413 @@
+
+    <MenuPopover>
+      <MenuList>
+        <MenuItem>New</MenuItem>
+        <MenuItem>New Window</MenuItem>
+        <MenuItem disabled>Open File</MenuItem>
+        <MenuItem>Open Folder</MenuItem>
+      </MenuList>
+    </MenuPopover>
+  </Menu>
+);
+
+// ------------------------------ Sticky tree implementation ------------------------------
+
+const useStyles = makeStyles({
+  rail: {
+    width: '360px',
+    minWidth: '360px',
+    height: '700px',
+    // boxSizing: 'border-box',
+    ...shorthands.border('1px', 'solid', 'black'),
+    overflowY: 'auto',
+    overflowX: 'hidden',
+  },
+  tree: {
+    rowGap: '0px',
+  },
+  header: {
+    backgroundColor: 'pink',
+    zIndex: 99,
+    position: 'sticky',
+    boxSizing: 'border-box',
+    ...shorthands.borderBottom('1px', 'solid', 'magenta'),
+  },
+});
+
+const RenderHeader = React.forwardRef<
+  HTMLDivElement,
+  {
+    item: HeadlessTreeItem<FlatItem>; // TODO type
+    headerHeight: number;
+  }
+>(({ item, headerHeight }, ref) => {
+  const classes = useStyles();
+  const focusTargetAttribute = useRestoreFocusTarget();
+
+  const { content, ...treeItemProps } = item.getTreeItemProps();
+  const pos = treeItemProps['aria-posinset'];
+  const setSize = treeItemProps['aria-setsize'];
+
+  return (
+    <TreeItem
+      aria-description="has context menu"
+      {...focusTargetAttribute}
+      {...treeItemProps}
+      className={classes.header}
+      style={{
+        top: `${headerHeight * (pos - 1)}px`,
+        bottom: `${headerHeight * (setSize - pos)}px`,
+        height: `${headerHeight}px`,
+      }}
+      ref={ref}
+    >
+      <TreeItemLayout actions={<ActionsExample />}>{content}</TreeItemLayout>
+    </TreeItem>
+  );
+});
+
+const useScrollItemIfNeeded_item = ({
+  value,
+  parentJustExpanded,
+  isBehindHeaders,
+}: {
+  value: TreeItemValue;
+  parentJustExpanded?: boolean;
+  isBehindHeaders: (element: HTMLElement, value: TreeItemValue) => boolean | undefined;
+}) => {
+  const itemRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    if (itemRef.current && parentJustExpanded) {
+      // Issue1: solution 1 - this is the simplest solution.
+      // Pro: simple
+      // Con:
+      // - scroll jump for a collapsed header in the middle.
+      // - No guarantee of visibility.
+      // itemRef.current.scrollIntoView({ block: 'center' });
+      //
+      // Issue1: solution 2 - check if the item is hidden, if it is, scroll it to center
+      // Pro: no scroll jump
+      // Con:
+      // - need to keep ref on all headers
+      // - no gurantee of visibility
+      if (isBehindHeaders(itemRef.current, value)) {
+        itemRef.current.scrollIntoView({ block: 'center' }); // TODO: scrollIntoView scrolls everything, so if the container is in another scrollable container, it may scroll the container as well.
+      }
+      //
+      // Issue1: solution 3 - check if the item is hidden, if it is, dock its parent to top?
```

Comment:
Thanks for keeping comments on your previous solutions and their pros/cons. This is helpful for the preview.

If you plan on removing these before merging, could you please include them in a MD file in this directory or an RFC somewhere for future reference?